core/src/main/java/com/sequenceiq/cloudbreak/service/salt/PartialSaltStateUpdateService.java

```diff
@@ -0,0 +1,63 @@
+package com.sequenceiq.cloudbreak.service.salt;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.exception.CloudbreakServiceException;
+import com.sequenceiq.cloudbreak.orchestrator.host.StateUpdateService;
+import com.sequenceiq.cloudbreak.orchestrator.metadata.OrchestratorMetadata;
+import com.sequenceiq.cloudbreak.orchestrator.metadata.OrchestratorMetadataProvider;
+import com.sequenceiq.cloudbreak.util.CompressUtil;
+
+@Service
+public class PartialSaltStateUpdateService {
+    private static final String SALT_DIRECTORY_PREFIX = "/salt/";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PartialSaltStateUpdateService.class);
+
+    @Inject
+    private OrchestratorMetadataProvider orchestratorMetadataProvider;
+
+    @Inject
+    private CompressUtil compressUtil;
+
+    @Inject
+    private StateUpdateService stateUpdateService;
+
+    public void performSaltUpdate(Long stackId, List<String> componentNames, Set<String> additionalTargetPrivateIps, Set<String> additionalTargetHostNames) {
+        List<String> saltStateDefinitions = orchestratorMetadataProvider.getSaltStateDefinitionBaseFolders();
+        List<String> filteredSaltComponents = componentNames.stream().map(name -> SALT_DIRECTORY_PREFIX + name).collect(Collectors.toList());
+        LOGGER.info("Performing partial salt update for Salt components: {}", filteredSaltComponents);
+        try {
+            byte[] currentSaltState = orchestratorMetadataProvider.getStoredStates(stackId);
+            byte[] telemetrySaltStateConfigs = compressUtil.generateCompressedOutputFromFolders(saltStateDefinitions, filteredSaltComponents);
```

Comment:
`telemetrySaltStateConfigs` this variable name does not look correct

Reply:
Ah you are correct, thank you. Some of this is copied from somewhere else.